packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+        ]);
+    });
+
+    xit('dispatches `rangeChanged` events on Virtualized Table', async () => {
+        // console.log('This test');
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        // console.log(1);
+        await nextFrame();
+        // console.log(2);
+        await nextFrame();
+        // console.log(3);
+        await nextFrame();
+        // debugger;
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        // console.log('4a');
+        const rangeChanged = oneEvent(el, 'rangeChanged');
+        // console.log(4);
+        let tableRow = el.querySelector('sp-table-row') as TableRow;
+        const initialValue = tableRow.value;
+        el.scrollToIndex(47);
+        // (el as unknown as { tableBody: TableBody }).tableBody.scrollTop = (el as unknown as { tableBody: TableBody }).tableBody.scrollHeight;
+        // console.log('we await after this...');
+        await nextFrame();
+        await nextFrame();
+        // debugger;
+        await rangeChanged;
+        tableRow = el.querySelector('sp-table-row') as TableRow;
+        const newValue = tableRow.value;
+
+        expect(newValue).to.not.equal(initialValue);
+    });
+
+    it('dispatches `visibilityChanged` events on Virtualized Table', async () => {
+        // console.log('NOT this test');
+        const visibilityChangedSpy = spy();
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+                @visibilityChanged=${({
+                    target,
+                }: Event & { target: Table }) => {
+                    visibilityChangedSpy(target);
+                }}
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
```

Comment:
I vaguely remember this coming because we need to wait for Table, then Table Body and the Virtualizer, but to help future you please comment anytime we use more than one `await` to ensure the browser is ready for the next step of a test.